include/psp2/jpegencarm.h

```diff
@@ -1,101 +1,165 @@
+
+/**
+ * JPEG Header Modes
+ */
+typedef enum SceJpegArmEncoderHeaderMode {
+	SCE_JPEGENCARM_HEADER_MODE_JPEG  = 0,  //!< JPEG header mode
+	SCE_JPEGENCARM_HEADER_MODE_MJPEG = 1   //!< MJPEG header mode
+} SceJpegArmEncoderHeaderMode;
+
+/**
+ * Get required size of context memory.
+ *
+ * @return Required size of allocated memory.
+ */
+SceSize sceJpegArmEncoderGetContextSize(void);
+
+/**
+ * Initialize a JPEG encoder.
  *
- * @param[in] context - A pointer to a big enough allocated memory block
- * @param[in] inWidth - Input width in pixels
- * @param[in] inHeight - Input height in pixels
- * @param[in] pixelformat - A valid ::SceJpegEncoderPixelFormat set of values
- * @param[in] outBuffer - A physically continuous memory block 256 bytes aligned
- * @param[in] outSize - Output size in bytes
+ * @param[in] context     - An allocated encoder context of appropriate size.
+ * @param[in] inWidth     - Input width in pixels.
+ * @param[in] inHeight    - Input height in pixels.
+ * @param[in] pixelformat - One of ::SceJpegArmEncoderPixelFormat.
+ * @param[in] outBuffer   - A sufficiently sized 8 byte aligned output buffer.
+ * @param[in] outSize     - Output buffer size in bytes.
  *
  * @return 0 on success, < 0 on error.
  */
-int sceJpegArmEncoderInit(SceJpegArmEncoderContext context, int inWidth, int inHeight, SceJpegEncoderPixelFormat pixelformat, void *outBuffer, SceSize outSize);
+int sceJpegArmEncoderInit(SceJpegArmEncoderContext context, SceUInt16 inWidth, SceUInt16 inHeight, SceJpegArmEncoderPixelFormat pixelformat, void *outBuffer, SceSize outSize);
 
 /**
- * Terminate a jpeg encoder
+ * Terminate a JPEG encoder.
  *
- * @param[in] context - A pointer to an already initialized ::SceJpegArmEncoderContext
+ * @param[in] context - An already initialized ::SceJpegArmEncoderContext.
  *
  * @return 0 on success, < 0 on error.
  */
 int sceJpegArmEncoderEnd(SceJpegArmEncoderContext context);
 
 /**
- * Execute a jpeg encode
+ * Execute a JPEG encode.
  *
- * @param[in] context - A pointer to an already initialized ::SceJpegArmEncoderContext
- * @param[in] inBuffer - A physically continuous memory block 256 bytes aligned
+ * @param[in] context  - An already initialized ::SceJpegArmEncoderContext.
+ * @param[in] inBuffer - An 8 byte aligned memory block of color data.
  *
- * @return encoded jpeg size on success, < 0 on error.
+ * @return Encoded JPEG size on success, < 0 on error.
  */
 int sceJpegArmEncoderEncode(SceJpegArmEncoderContext context, const void *inBuffer);
 
 /**
- * Set encoder compression ratio
+ * Set the encoder compression ratio.
  *
- * @param[in] context - A pointer to an already initialized ::SceJpegArmEncoderContext
- * @param[in] ratio - A value between 0 and 255 (higher = better compression, lower = better speed)
+ * @param[in] context - An already initialized ::SceJpegArmEncoderContext.
+ * @param[in] ratio   - A value between 1 and 256 (higher = better compression, lower = better speed).
```

Comment:
2 to 255?

Reply:
Fixed, was a mistake from early testing.